pyproject.toml

```diff
@@ -32,3 +32,67 @@ include = ["PyCO2SYS*"]
 
 [tool.setuptools.dynamic]
 version = {attr = "PyCO2SYS.meta.version"}
+
+
+[tool.ruff.lint]
+select = [
+    "D",  # pydocstyle
+    "E",  # Error
+    "F",  # pyflakes
+    "I",  # isort
+    "B",  # Bugbear
+    "UP", # pyupgrade
+    "LOG", # logging
+    "ICN", # import conventions
+    "G", # logging-format
+    "RUF", # ruff
+    "ISC001", # single-line-implicit-string-concatenation
+]
+
+ignore = [
+    # line too long (82 > 79 characters)
+    "E501",
+    # Missing docstring in public module
+    "D100",
+    # Missing docstring in public class
+    "D101",
+    # Missing docstring in public method
+    "D102",
+    # Missing docstring in public function
+    "D103",
+    # Missing docstring in public package
+    "D104",
+    # Missing docstring in magic method
+    "D105",
+    # Missing docstring in __init__
+    "D400",
+    # First line should be in imperative mood (requires writing of summaries)
+    "D401",
+    # First word of docstring is uncapitalized
+    "D403",
+    # First word of the docstring should not be `This`
+    "D404",
+    # 1 blank line required between summary line and description (requires writing of summaries)
+    "D205",
+    # Docstring contains ambiguous...
+    "RUF002",
+
+    # TODO: These should be fixed
+    "B006", # Do not use mutable data structures for argument defaults
+    "B007", # Loop control variable `funcs` not used within loop body
+    "B011", # Do not `assert False`
+    "B028", # No explicit `stacklevel` keyword argument found
+    "B904", # Within an `except` clause, raise exceptions
+    "RUF001", # String contains ambiguous ...
+    "E402", # Module level import not at top of file
+    "E711", # Comparison to `None` should be `cond is None`
+    "E721", # Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks
+    "F401", # ... imported but unused; consider removing, adding to `__all__`, or using a redundant alias
+    "ICN", # import conventions
```

Comment:
In my opinion, these would be ones to consider fixing. More info can be found by googling "{code} ruff". If you feel some of these aren't worth fixing, they can be moved out of the "todo" block

Btw, one can ignore a rule for a line if you want to intentionally do something. see [error suppression | Ruff docs](https://docs.astral.sh/ruff/linter/#error-suppression)